File: java/src/main/java/pro/wata/sdk/model/Transaction.java
package pro.wata.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.time.OffsetDateTime;

/**
 * Транзакция. Обратите внимание: здесь комиссия называется
 * {@code totalCommission}, тогда как в вебхуке то же значение приходит как
 * {@code commission} (см. {@link pro.wata.sdk.model.WebhookEvent#commission()}).
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Transaction(
        String id,
        TransactionType transactionType,
        TransactionKind kind,
        TransactionStatus status,
        Double amount,
        Currency currency,
        Double totalCommission,
        String orderId,
        String orderDescription,
        String paymentLinkId,
        String originalTransactionId,
        String errorCode,
        String errorDescription,
        OffsetDateTime creationTime,
        OffsetDateTime paymentTime
) {
}

File: dotnet/src/Wata.Sdk/Acquiring/Models/Transaction.cs
using Wata.Sdk.Models;

namespace Wata.Sdk.Acquiring.Models;

public sealed record Transaction
{
    public Guid Id { get; init; }

    public TransactionType Type { get; init; }

    public TransactionKind Kind { get; init; }

    public TransactionStatus Status { get; init; }

    public decimal Amount { get; init; }

    public Currency Currency { get; init; }

    public string? OrderId { get; init; }

    public string? OrderDescription { get; init; }

    public Guid? PaymentLinkId { get; init; }

    public string? TerminalPublicId { get; init; }

    public string? TerminalName { get; init; }

    /// <summary>Комиссия. В вебхуке то же значение называется commission (см. README).</summary>
    public decimal? TotalCommission { get; init; }

    public DateTimeOffset CreationTime { get; init; }

    public DateTimeOffset? PaymentTime { get; init; }

    public string? ErrorCode { get; init; }

    public string? ErrorDescription { get; init; }
}
